site_scons/prereq_tools/base.py

```diff
@@ -1418,7 +1418,11 @@ def set_environment(self, env, needed_libs):
 
     def _set_build_env(self, env):
         """Add any environment variables to build environment"""
-        env["ENV"].update(self.key_words.get("build_env", {}))
+        build_env = self.key_words.get("build_env", {})
+        newenv = {}
+        for key, value in build_env.items():
```

Comment:
why do we have to do this manually, is there no built-in to copy env?

Reply:
I need to do subst on each entry because I have stuff like $SPDK_PREFIX in there